node/lib/package/LocalPackageSource.ts

```diff
@@ -319,14 +320,112 @@ export class LocalPackageSource extends PackageSource {
+                    // By convention the schema module file and its parent directory
+                    // both match the name of the schema.
+                    let schemaModulePath: string = packageJson.name + "/" + schemaId;
+                    schemaModulePaths.push(schemaModulePath); //TODO: validate this
+                    schemaInfos.push({
+                        moduleName: schemaModulePath
+                    });
+                }
+            });
+        }
+
+        // TODO: if we are getting rid of <arg> it should reflect in transalator parsing as well.
+        // Parse onboarding info from the manifest.
+        let onboardingModulePath: string = "";
+        let onboardingInfo: any[] = [];
+        let onboardingFlowElements: any[] = [];
+        if (Array.isArray(manifestXmlRoot.onboarding) &&
+                manifestXmlRoot.onboarding.length == 1) {
+            let onboardingElement: any = manifestXmlRoot.onboarding[0];
+            let onboardingId:string = onboardingElement.$.id;
+            if (onboardingId) {
+                // By convention the onboarding module is in a package whose name is derived
+                // from the onboarding id.
+                let onboardingPackageName: string =
+                        LocalPackageSource.derivePackageName(onboardingId, "onboarding"); // TODO check this
+                onboardingModulePath = onboardingPackageName + "/" + onboardingId + "/" + onboardingId;
+                
+                 if (Array.isArray(onboardingElement.onboardflow)) {
+                    onboardingElement.onboardflow.forEach((onboardFlowElement: any) => {
+                        let flowElements: any[] = [];
+                        let onboardFlowName: string = onboardFlowElement.$.name;
+                        
+                        if(Array.isArray(onboardFlowElement.flow)){
+                            onboardFlowElement.flow.forEach((flowElement: any) => {
+                                let descriptionProperties: any = {};
+                                flowElement.description.forEach((descriptionElement: any) => {
+                                    descriptionProperties[descriptionElement.$.language] = descriptionElement._;
+                                }); // <description>
+                                flowElements.push({
+                                    type: flowElement.$.type,
+                                    name: flowElement.name[0],
+                                    descriptions: descriptionProperties
+                                });
+                            }); // <flow>
+                        }
+                        onboardingFlowElements.push({
+                            name: onboardFlowName,
+                            flow: flowElements
+                        });
+                    }); // <onboardflow>
+                }
+            }
+            onboardingInfo.push({
+                onboardingId : onboardingId,
```

Comment:
I'm referring to the onboarding module path. Following the pattern in the other classes, it would be PackageOnboardingInfo.moduleName. (I think of a module *path* as relative to the application, while a module name is relative to the package root.)

---
In reply to: [75344669](https://github.com/openT2T/opent2t/pull/22#discussion_r75344669) [](ancestors = 75344669,75264826)